File: test/fixtures/params/ios/Sources/ParamsPlugin.swift
import Foundation
import Capacitor

@objc(ParamsPlugin)
public class ParamsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "ParamsPlugin"
    public let jsName = "Params"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "get", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "set", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "configure", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "keys", returnType: CAPPluginReturnPromise)
    ]

    private var store: [String: String] = [:]

    @objc func get(_ call: CAPPluginCall) {
        guard let key = call.getString("key") else {
            call.reject("key required")
            return
        }
        call.resolve([
            "value": store[key] as Any
        ])
    }

    @objc func set(_ call: CAPPluginCall) {
        guard let key = call.getString("key") else {
            call.reject("key required")
            return
        }
        let value = call.getString("value", "")
        store[key] = value
        call.resolve()
    }

    @objc func configure(_ call: CAPPluginCall) {
        let group = call.getString("group")
        if let group = group {
            print("group \(group)")
        }
        call.resolve()
    }

    @objc func keys(_ call: CAPPluginCall) {
        call.resolve([
            "keys": Array(store.keys)
        ])
    }
}

File: test/fixtures/params/android/src/main/java/com/example/params/ParamsPlugin.java
package com.example.params;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Params")
public class ParamsPlugin extends Plugin {

    private java.util.Map<String, String> store = new java.util.HashMap<>();

    @PluginMethod
    public void get(PluginCall call) {
        String key = call.getString("key");
        if (key == null) {
            call.reject("key required");
            return;
        }
        JSObject ret = new JSObject();
        ret.put("value", store.get(key));
        call.resolve(ret);
    }

    @PluginMethod
    public void set(PluginCall call) {
        String key = call.getString("key");
        if (key == null) {
            call.reject("key required");
            return;
        }
        String value = call.getString("value", "");
        store.put(key, value);
        call.resolve();
    }

    @PluginMethod
    public void configure(PluginCall call) {
        String group = call.getString("group");
        call.resolve();
    }

    @PluginMethod
    public void keys(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("keys", new java.util.ArrayList<>(store.keySet()));
        call.resolve(ret);
    }
}
